odoo/addons/base/models/ir_topbar_actions.py

```diff
@@ -0,0 +1,74 @@
+    # It is required to have either action_id or python_action
+    action_id = fields.Many2one('ir.actions.act_window', string="Action Id", ondelete="cascade")
+    python_action = fields.Char(string="Python Action")
+    user_id = fields.Many2one('res.users', string="User specific topbar action. If empty, shared topbar action", ondelete="cascade")
+    is_deletable = fields.Boolean(compute="_compute_is_deletable")
+    default_view_mode = fields.Char(string="Default view (if none, default view of the action is taken)")
+    filter_ids = fields.One2many("ir.filters", "topbar_action_id", string="Default filter of the topbar action (if none, no filters)")
+    is_visible = fields.Boolean(string="Computed field to check if the record should be visible acording to the domain", compute="_compute_is_visible")
+    domain = fields.Char(string='Domain Value', default="[]",
+                         help="Domain applied to the current Id of the Parent Model")
+    context = fields.Char(string='Context Value', default={}, required=True,
+                          help="Context dictionary as Python expression, empty by default (Default: {})")
+
+    @api.constrains('action_id', 'python_action')
+    def _check_only_one_action_defined(self):
+        for record in self:
+            if bool(record.action_id) == bool(record.python_action):
+                raise ValidationError("You cannot define a xml action and a python action for the same topbar action.")
+            elif record.python_action and not record.name:
+                raise ValidationError("You cannot define a xml python action and not define the name.")
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if "action_id" in vals and isinstance(vals["action_id"], list):
+                vals["action_id"] = vals["action_id"][0]
+            if "parent_action_id" in vals and isinstance(vals["parent_action_id"], list):
+                vals["parent_action_id"] = vals["parent_action_id"][0]
+            if "name" not in vals:
+                vals["name"] = self.env["ir.actions.act_window"].browse(vals["action_id"]).name
+        return super().create(vals_list)
+
+    def _compute_is_deletable(self):
+        external_ids = self._get_external_ids()
+        for record in self:
+            record_external_ids = external_ids[record.id]
+            record["is_deletable"] = not any(
+                not ex_id.startswith(("__export__", "__custom__")) for ex_id in record_external_ids
+            )
+
+    def _compute_is_visible(self):
+        if not self:
+            return
+        active_id = self.env.context.get("active_id", False)
+        domain_id = [("id", "=", active_id)] if active_id else []
+        for record in self:
+            domain_model = literal_eval(record.domain)
+            record.is_visible = self.env[record.res_model].search_count(domain_id + domain_model) > 0
```

Comment:
```suggestion
            record.is_visible = self.env[record.res_model].search_count(domain.AND([domain_id, domain_model]))
```
- `domain.AND`
- I guess the fact that the field is a boolean casts it automatically